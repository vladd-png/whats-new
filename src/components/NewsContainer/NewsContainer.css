.newsContainer {
  display: flex;
  flex-direction: row;
  width: 100%;
  overflow: scroll;
  flex-wrap: wrap;
  justify-content: center;
}
