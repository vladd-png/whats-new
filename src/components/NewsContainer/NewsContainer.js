import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle.js';

const NewsContainer = (props) => {
  const newsType = props.newsType.map(news => {
    return <NewsArticle
      key={news.id}
      headline={news.headline}
      body={news.description}
      img={news.img}
      link={news.url}
    />
  })
  return (
    <section className='newsContainer'>
      {newsType}
    </section>
  )
}

export default NewsContainer;
