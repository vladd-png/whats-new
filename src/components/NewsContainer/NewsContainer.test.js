import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {

  it('should be an instance of NewsContainer component', () => {
    const wrapper = shallow(<NewsContainer
        newsType={[{
          headline: 'Ramen',
          description: 'Oriental Chicken Flavor',
          url: 'url.link',
          id: 7,
          img: 'ur.link',
        }]}
      />);
      expect(wrapper).toMatchSnapshot();
  });

});
