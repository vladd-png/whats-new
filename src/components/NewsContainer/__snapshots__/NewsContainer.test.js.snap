// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should be an instance of NewsContainer component 1`] = `
<section
  className="newsContainer"
>
  <NewsArticle
    body="Oriental Chicken Flavor"
    headline="Ramen"
    img="ur.link"
    key="7"
    link="url.link"
  />
</section>
`;
